R/logistics_helper.R

```diff
@@ -0,0 +1,72 @@
+## log-likelihood
+logit_loglik <- function(beta, X, Y){
+  p <- 1 / (1 + exp(-X %*% beta))
+  l <- -(t(Y) %*% log(p) + t(1 - Y) %*% log(1 - p))
```

Comment:
You implemented the negative log likelihood (and respective gradients/hessians), so your BFGS is trying to maximize the wrong thing.